packaging/docker/fdb-aws-s3-credentials-fetcher/fdb-aws-s3-credentials-fetcher.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+)
+
+type BlobCredentials struct {
+    Accounts map[string]Account `json:"accounts"`
+}
+
+type Account struct {
+    Secret  string `json:"secret"`
+    APIKey  string `json:"api_key"`
+    Token   string `json:"token"`
+}
+
+func writeCredentialsFile(bucket, region, credFile string, accessKey, secretKey, token string) error {
+    s3Endpoint := fmt.Sprintf("s3.%s.amazonaws.com", region)
+    
+    // Bulkload hostname is like this backup-112664522426-us-west-2.s3-us-west-2.amazonaws.com
+    // i.e. bucket and then s3.region.amazonaws.com.
+    // So add records for this format and for s3.REGION.amazonaws.com too.
+    blobCreds := BlobCredentials{
+        Accounts: map[string]Account{
+            "@" + s3Endpoint: {
+                Secret:  secretKey,
+                APIKey:  accessKey,
+                Token:   token,
+            },
+            "@" + s3Endpoint + ":443": {
+                Secret:  secretKey,
+                APIKey:  accessKey,
+                Token:   token,
+            },
+            "@" + bucket + "." + s3Endpoint: {
+                Secret:  secretKey,
+                APIKey:  accessKey,
+                Token:   token,
+            },
+            "@" + bucket + "." + s3Endpoint + ":443": {
+                Secret:  secretKey,
+                APIKey:  accessKey,
+                Token:   token,
+            },
+        },
+    }
+
+    data, err := json.Marshal(blobCreds)
+    if err != nil {
+        return fmt.Errorf("failed to marshal credentials: %v", err)
+    }
+
+    if err := os.WriteFile(credFile, data, 0644); err != nil {
+        return fmt.Errorf("failed to write credentials file: %v", err)
```

Comment:
```suggestion
        return fmt.Errorf("failed to write credentials file: %w", err)
```

See https://go.dev/blog/go1.13-errors (only if you're interested) :)